src/Service/RequestService.php

```diff
@@ -724,20 +771,52 @@ public function requestHandler(array $data, array $configuration): Response
             $appEndpointConfig = $this->getAppEndpointConfig();
         }
 
-        // Work around the _ with a custom function for getting clean query parameters from a request
-        $filters = $this->realRequestQueryAll();
+        // Make a list of schema's that are allowed for this endpoint.
+        $allowedSchemas = $this->getAllowedSchemas();
 
-        // Handle mapping for query parameters
-        if (isset($appEndpointConfig['in']['query']) === true) {
-            $filters = $this->queryAppEndpointConfig($filters, $appEndpointConfig['in']['query']);
+        // Check if the user has the needed scopes.
+        $securityResponse = $this->checkUserScopes($allowedSchemas['reference']);
+        if ($securityResponse instanceof Response === true) {
+            return $securityResponse;
         }
 
         // Get the ID.
         $this->identification = $this->getId();
 
         // If we have an ID we can get an Object to work with (except on gets we handle those from cache).
         if (isset($this->identification) === true && empty($this->identification) === false && $this->data['method'] != 'GET') {
-            $this->object = $this->entityManager->getRepository('App:ObjectEntity')->findOneBy(['id' => $this->identification]);
+            $object = $this->entityManager->getRepository('App:ObjectEntity')->findOneBy(['id' => $this->identification]);
+            if ($object === null) {
+                return new Response(
+                    $this->serializeData(
+                        [
+                            'message' => 'Could not find an object with id '.$this->identification,
+                            'type'    => 'Bad Request',
+                            'path'    => implode(', ', $allowedSchemas['name']),
+                            'data'    => ['id' => $this->identification],
+                        ],
+                        $contentType
+                    ),
+                    Response::HTTP_NOT_FOUND,
+                    ['Content-type' => $contentType]
+                );
+            }
+
+            $this->object = $object;
+        }
+
+        // Check if user is allowed to change this object (owner & organization). Checking if an object may be viewed (GET) is done in the CacheService
+        $securityResponse = $this->checkOwnerAndOrg();
+        if ($securityResponse instanceof Response === true) {
```

Comment:
If possible might want to do this before fetching the object? For small performance gain?

Reply:
We need the object in order to check if a User is allowed to edit it (PUT, PATCH, DELETE, this is the only case in which $this->object is set and we actually check for owner and org in this checkOwnerAndOrg function)

(GET calls are checked in the CacheService)